File: baekjoon-online-judge/silver/09372.py
from sys import stdin
input = stdin.readline

if __name__ == '__main__':
    # Input
    res = []
    for _ in range(int(input())):
        N, M = map(int, input().split())
        for _ in range(M):
            input()

        # Solve
        res.append(N - 1)

    # Output
    print(*res, sep='\n')

File: baekjoon-online-judge/silver/09372.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        StringBuilder sb = new StringBuilder();

        // Input
        int T = Integer.parseInt(br.readLine());
        while (T-- > 0) {
            st = new StringTokenizer(br.readLine());
            int N = Integer.parseInt(st.nextToken());
            int M = Integer.parseInt(st.nextToken());
            while (M-- > 0) {
                br.readLine();
            }

            // Solve
            sb.append(N - 1).append('\n');
        }

        // Output
        System.out.print(sb);
    }
}